ui/src/app/machines/views/MachineDetails/MachineConfiguration/PowerForm/PowerForm.test.tsx

```diff
@@ -0,0 +1,142 @@
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+
+import PowerForm from "./PowerForm";
+
+import type { RootState } from "app/store/root/types";
+import {
+  generalState as generalStateFactory,
+  machineDetails as machineDetailsFactory,
+  machineState as machineStateFactory,
+  machineStatus as machineStatusFactory,
+  machineStatuses as machineStatusesFactory,
+  powerField as powerFieldFactory,
+  powerType as powerTypeFactory,
+  powerTypesState as powerTypesStateFactory,
+  rootState as rootStateFactory,
+} from "testing/factories";
+
+const mockStore = configureStore();
+
+describe("PowerForm", () => {
+  let state: RootState;
+
+  beforeEach(() => {
+    state = rootStateFactory({
+      general: generalStateFactory({
+        powerTypes: powerTypesStateFactory({
+          data: [powerTypeFactory({ fields: [], name: "manual" })],
+          loaded: true,
+        }),
+      }),
+      machine: machineStateFactory({
+        items: [machineDetailsFactory({ system_id: "abc123" })],
+        statuses: machineStatusesFactory({
+          abc123: machineStatusFactory(),
+        }),
+      }),
+    });
+  });
+
+  it("is not editable if machine does not have edit permission", () => {
+    state.machine.items[0].permissions = [];
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <PowerForm systemId="abc123" />
+      </Provider>
+    );
+
+    expect(wrapper.find("button[data-test='edit-power-config']").exists()).toBe(
+      false
+    );
+  });
+
+  it("is disabled until the edit button is pressed", () => {
+    state.machine.items[0].permissions = ["edit"];
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <PowerForm systemId="abc123" />
+      </Provider>
+    );
+
+    expect(wrapper.find("FormikForm").prop("editable")).toBe(false);
+
+    wrapper.find("button[data-test='edit-power-config']").simulate("click");
+
+    expect(wrapper.find("FormikForm").prop("editable")).toBe(true);
```

Comment:
Looks

like

this

could

probably

have

less

empty

lines.

Reply:
But





why?